tests/src/testhyperelasticity.hh

```diff
@@ -0,0 +1,252 @@
+// SPDX-FileCopyrightText: 2021-2025 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#include <config.h>
+
+#include "materialresultcollection.hh"
+#include "testhelpers.hh"
+
+#include <dune/common/test/testsuite.hh>
+
+#include <Eigen/Eigenvalues>
+
+#include <autodiff/forward/dual.hpp>
+#include <autodiff/forward/dual/dual.hpp>
+#include <autodiff/forward/dual/eigen.hpp>
+#include <autodiff/forward/utils/derivative.hpp>
+
+#include <ikarus/finiteelements/mechanics/materials.hh>
+#include <ikarus/finiteelements/physicshelper.hh>
+#include <ikarus/utils/functionsanitychecks.hh>
+#include <ikarus/utils/nonlinearoperator.hh>
+
+using namespace Ikarus;
+using namespace Materials;
+using Dune::TestSuite;
+
+inline auto testMatrix() {
+  return Eigen::Matrix3d{
+      { 0.600872, -0.179083, 0},
+      {-0.179083,  0.859121, 0},
+      {        0,         0, 1}
+  };
+}
```

Comment:
Ah or add here a test matrix with reapearing eigenvalues

Reply:
The eigenvalues are the same for the undeformed case, isn't it? That's the reason it was added